sql/analyzer/apply_foreign_keys.go

```diff
@@ -160,57 +165,249 @@ func applyForeignKeysToNodes(ctx *sql.Context, a *Analyzer, n sql.Node, cache *f
 	}
 }
 
+// getForeignKeyEditor merges both getForeignKeyReferences and getForeignKeyRefActions and returns a single editor.
+func getForeignKeyEditor(ctx *sql.Context, a *Analyzer, tbl sql.ForeignKeyTable, cache *foreignKeyTableCache, fkChain foreignKeyChain) (*plan.ForeignKeyEditor, error) {
+	fkEditor, err := getForeignKeyReferences(ctx, a, tbl, cache, fkChain)
+	if err != nil {
+		return nil, err
+	}
+	fkEditor2, err := getForeignKeyRefActions(ctx, a, tbl, cache, fkChain)
+	if err != nil {
+		return nil, err
+	}
+	// Due to tables not containing their database, we have to grab the table's database from the foreign key definition.
```

Comment:
I don't really follow what's happening here. Could we just add a `Database()` method to `sql.ForeignKeyTable`?